### PR TITLE
Refactor/smart pointers scene

### PR DESCRIPTION
**Pull request by Jonazan2**

### CODE REVIEW

**Comment by Jonazan2** [comment]
Something really weird is happening with this merge request. I think we should merge it manually via command line.

File: Dominus/Core/Engine/GLFWInputHandler.cpp [hunk 1]
```diff
@@ -114,7 +114,7 @@ void GLFWInputHandler::init() {
     glfwSetWindowCloseCallback( window, window_close_callback );
 }
 
-std::vector<Event*> * GLFWInputHandler::poolEvents() {
+std::vector< std::shared_ptr< Event > > GLFWInputHandler::poolEvents() {
     glfwPollEvents ();
-    return &events;
+    return events;
```
Comment (AlvaroChambi) [inline Dominus/Core/Engine/GLFWInputHandler.cpp:119]: Why return the value and not the reference?

File: Headers/Core/Engine/InputHandler.h [hunk 2]
```diff
@@ -1,21 +1,15 @@
...
+#include <vector>
+#include <memory>
 
 #include "Event.h"
-#include <vector>
 
 class InputHandler {
 public:
-    virtual std::vector<Event*> * poolEvents() = 0;
+    virtual std::vector< std::shared_ptr< Event > > poolEvents() = 0;
```
Comment (AlvaroChambi) [inline Headers/Core/Engine/InputHandler.h:11]: Here we need to return the vector reference

File: Headers/Core/Scenes/GameScene.h [hunk 3]
```diff
@@ -1,22 +1,17 @@
...
 #include "Scene.h"
-#include <vector>
 #include "Event.h"
 
 class GameScene {
 public:
     virtual void onSceneCreated( Scene* scene ) = 0;
     virtual void onUpdate( double delta ) = 0;
-    virtual void onCosumeInput( std::vector<Event*>* ) = 0;
+    virtual void onCosumeInput( std::vector< std::shared_ptr< Event > > ) = 0;
```
Comment (AlvaroChambi) [inline Headers/Core/Scenes/GameScene.h:14]: the same, here we need the vector reference

File: Headers/Core/Scenes/RenderScene.h [hunk 4]
```diff
@@ -18,14 +10,15 @@ class RenderScene : public GameScene {
     
     void onUpdate( double delta );
     void onSceneCreated( Scene* scene );
-    void onCosumeInput( std::vector<Event*>* );
+    void onCosumeInput( std::vector< std::shared_ptr< Event > > );
```
Comment (AlvaroChambi) [inline Headers/Core/Scenes/RenderScene.h:13]: the same, here we need the vector reference

File: Dominus/Core/Engine/Engine.cpp [hunk 5]
```diff
@@ -30,15 +30,14 @@ void Engine::init() {
...
+    std::vector< std::shared_ptr< Event > > events = inputHandler->poolEvents();
     consumeEvents( events );
     gameScene->onCosumeInput( events );
-    events->clear();
 }
 
-void Engine::consumeEvents( std::vector<Event *> * events ) {
-    for ( int i = 0; i < events->size(); i++ ) {
-        Event* event = events->at( i );
+void Engine::consumeEvents( std::vector< std::shared_ptr< Event > > events ) {
```
Comment (AlvaroChambi) [inline Dominus/Core/Engine/Engine.cpp:38]: the same, here we need the vector reference

**Pull request closed without merging**